app/models/organisation.rb

```diff
@@ -25,6 +25,11 @@ def affiliated_stock_owners
       ids << id
     end
 
+    absorbed_organisations.each do |organisation|
+      ids.concat(organisation.stock_owners.pluck(:id))
+      ids << organisation.id if organisation.holds_own_stock?
```

Comment:
isn't this condition always true?

Reply:
What makes you think that? I think merged organisation might not hold stock (?) Am I missing something?

Reply:
Ah no you're right, I was still thinking we were looking at stock owners of this org, ignore me